src/core/probe/kernel/kernel.rs

```diff
@@ -59,33 +60,90 @@ impl fmt::Display for KernelProbe {
     }
 }
 
-/// Registers the unmarshaler for the kernel section of the event.
-pub(crate) fn register_unmarshaler(events: &mut BpfEvents) -> Result<()> {
-    events.register_unmarshaler(
-        BpfEventOwner::Kernel,
-        Box::new(
-            |raw_section: &BpfRawSection, fields: &mut Vec<EventField>| {
-                if raw_section.data.len() != 9 {
-                    bail!(
-                        "Section data is not the expected size {} != 9",
-                        raw_section.data.len()
-                    );
+struct KernelUnmarshaler {
+    #[cfg(not(test))]
+    stack_map: libbpf_rs::Map,
+}
+
+impl KernelUnmarshaler {
+    #[cfg(not(test))]
+    fn unmarshal_stackid(
+        &self,
+        raw_section: &BpfRawSection,
+        fields: &mut Vec<EventField>,
+    ) -> Result<()> {
+        let stackid = i64::from_ne_bytes(raw_section.data[9..17].try_into()?) as i32;
+        if stackid >= 0 {
+            let mut stack_trace: Vec<String> = Vec::new();
+            if let Some(stack_bytes) = self
+                .stack_map
+                .lookup(&stackid.to_ne_bytes(), libbpf_rs::MapFlags::ANY)?
+            {
+                let sstack: &[u64] = unsafe {
+                    std::slice::from_raw_parts(
+                        stack_bytes.as_ptr() as *const u64,
+                        stack_bytes.len() / std::mem::size_of::<u64>(),
+                    )
+                };
+
+                for sym in sstack {
+                    if *sym == 0x00_u64 {
+                        break;
+                    }
+
+                    match inspect::get_name_from_addr_near(*sym) {
```

Comment:
Maybe not for this time either but a symbol->name cache would be beneficial as it's very likely we get a few stack symbols repeated many times.

Reply:
it sure makes sense. Adding it in a separate patch should be fine.